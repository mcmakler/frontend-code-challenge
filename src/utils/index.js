import priceFormatter from './price-formatter/price-formatter';
import trim from './trim/trim';
import round from './round/round';
import carder from './carder/carder';
import flow from './flow/flow';
import * as specialCharacter from './special-characters/special-characters';
import * as mapper from './mapper/mapper';

export {
  priceFormatter,
  specialCharacter,
  mapper,
  trim,
  round,
  flow,
  carder
};
