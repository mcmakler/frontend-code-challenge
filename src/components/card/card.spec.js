import React from 'react';
import Card from './card';
import { mapper, carder }  from '../../utils';
import ReactTestRenderer from 'react-test-renderer';
import advertisementObj from '../../../fixtures/advertisement-fixture';

describe('Card specs', () => {
  it('should match snapshot', () => {
    const item = mapper.itemMapper(advertisementObj);
    const props = carder(item);
    const component = ReactTestRenderer.create(<Card {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
