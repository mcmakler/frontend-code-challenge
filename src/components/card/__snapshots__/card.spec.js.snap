// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card specs should match snapshot 1`] = `
<div
  className="col-xs-12 col-sm-6 col-md-4 card"
>
  <div
    className="box card__box"
  >
    <div
      className="cardHeader"
    >
      <img
        alt="FLATSCREEN ZUM EINZUG Moderne, frisch sanierte 3"
        className="cardHeader__img"
        src="https://d3e02gns9oqhhr.cloudfront.net/assets/expose_v2/573c7c608ffd7c0b2a8b45b6_variant_asset57ffa0ced86b1_inventoryM.jpg"
      />
      <span
        className="cardHeader__tag"
      >
        mieten
      </span>
    </div>
    <div
      className="cardTitle"
    >
      FLATSCREEN ZUM EINZUG Moderne, frisch sanierte 3
    </div>
    <div
      className="cardText"
    >
      Im Bonnefeld 7, 47259 Duisburg
    </div>
    <div
      className="cardFooter"
    >
      <div
        className="cardFooter__item--left auto-right-margin"
      >
        <span
          className="cardFooter__item__text cardFooter__item__text--bold "
        >
           
          374
           
          â‚¬
        </span>
      </div>
      <div
        className="cardFooter__item--right"
      >
        <span
          className="cardFooter__item__text  cardFooter__item__text--separated"
        >
           
          3
           
          Zimmer
        </span>
        <span
          className="cardFooter__item__text  cardFooter__item__text--separated"
        >
          ab
           
          79
           
          <span>
            m
            <span
              className="align-super font-10"
            >
              2
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
